course_catalog/views.py

```diff
@@ -328,3 +339,42 @@ class TopicViewSet(viewsets.ReadOnlyModelViewSet):
     serializer_class = CourseTopicSerializer
     pagination_class = LargePagination
     permission_classes = (AnonymousAccessReadonlyPermission,)
+
+
+class WebhookOCWView(APIView):
+    """
+    Handle webhooks coming from the OCW bucket
+    """
+
+    permission_classes = ()
+    authentication_classes = ()
+
+    def handle_exception(self, exc):
+        """Raise any exception with request info instead of returning response with error status/message"""
+        raise WebhookException(
+            f"REST Error ({exc}). BODY: {rapidjson.dumps(self.request.data or {})}, META: {self.request.META}"
+        ) from exc
+
+    def post(self, request):
+        """Process webhook request"""
+        if request.GET.get("webhook_key", None) != settings.OCW_WEBHOOK_KEY:
```

Comment:
Can we use [compare_digest](https://docs.python.org/3/library/hmac.html#hmac.compare_digest) here to avoid [this vulnerability](https://codahale.com/a-lesson-in-timing-attacks/) with short-circuit comparison?